src/common/exponent/xdlInterface.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {CommandExecutor, CommandVerbosity} from "../commandExecutor";
+import {Log} from "../log/log";
+
+import * as XDLPackage from "xdl";
+import * as Q from "q";
+
+const XDL_VERSION = "0.12.0";
+let xdlPackage: typeof XDLPackage;
+
+function getPackage(): Q.Promise<typeof XDLPackage> {
+    if (xdlPackage) {
+        return Q(xdlPackage);
+    }
+    // Don't do the require if we don't actually need it
+    try {
+        Log.logMessage("Getting exponent dependecy.", false);
+        xdlPackage = require("xdl");
+        return Q(xdlPackage);
+    } catch (e) {
+        // Ignore if not found
```

Comment:
Somehow when running in vscode 1.4.0 when xdl isn't installed and so the require above throws an exception, it isn't stopping at this catch block, but also the rejected promise handler that ends up catching this error doesn't actually get an error to catch, it just gets `undefined` for some reason.